standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/columnstats/merge/TimestampColumnStatsMerger.java

```diff
@@ -63,6 +64,18 @@ public void merge(ColumnStatisticsObj aggregateColStats, ColumnStatisticsObj new
       aggregateData.setNumDVs(ndv);
     }
 
+    if (aggregateData.getHistogramEstimator() != null && newData.getHistogramEstimator() != null) {
```

Comment:
Should we also take care of the case `aggregateData.getHistogramEstimator() == null` or `newData.getHistogramEstimator() == null` or even cannot merge?

Reply:
The case when `aggregateData.getHistogramEstimator() == null` but `newData.getHistogramEstimator() != null` should set `aggregateData` to `newData.getHistogramEstimator()`, this was missing, so thanks for noticing!

For the second case it's a no-op: if `newData.getHistogramEstimator() == null`, we just keep the one from the `aggregateData` (being it `null` or not).

I have added a new unit-test class (`ColumnStatsMergerTest`) to test all these cases, which can be run as follows:
```
mvn test -Dtest.groups=org.apache.hadoop.hive.metastore.annotation.MetastoreUnitTest -Dtest='ColumnStatsMergerTest.java' -pl standalone-metastore/metastore-server
```